app/controllers/katello/api/v2/products_controller.rb

```diff
@@ -43,6 +48,16 @@ def index_relation
       query = query.where(:id => @activation_key.products) if @activation_key
       query = query.where(:id => @system.products) if @system
       query = query.where(:id => Pool.find_by_id!(params[:subscription_id]).products) if params[:subscription_id]
+
+      # filter by sync plan
+      if sync_plan_id = params[:sync_plan_id]
+        query = if params[:available_for] == "sync_plan"
+                  query.where("sync_plan_id != ? OR sync_plan_id IS NULL", sync_plan_id)
```

Comment:
Yea, I was surprised at first too but [according to the Postgresql manual, any ordinary comparison operator that involves NULL returns NULL](http://www.postgresql.org/docs/current/interactive/functions-comparison.html). So for example `7 <> NULL` is NULL and not true.